torch/csrc/jit/tracer.h

```diff
@@ -64,34 +64,62 @@ inline std::vector<VariableFlags> getVarFlags(const variable_list& vars) {
 // are treated as constants.
 //
 // TODO: This code lives in the hotpath; make sure it is fast
-inline bool isTracing(const Variable& var) {
+inline bool isTracingVar(const Variable& var) {
   if (!var.defined() || !var.tracing_state()) return false;
   return std::any_of(var.tracing_state()->begin(), var.tracing_state()->end(), detail::isElemActive);
 }
 
-inline bool isTracing(std::initializer_list<Variable> vars) {
+inline bool isTracingVar(at::ArrayRef<Variable> vars) {
   // Reference to avoid refcount bump
-  for (auto& var : vars) {
-    if (isTracing(var)) return true;
+  for (const Variable& var : vars) {
+    if (isTracingVar(var)) return true;
   }
   return false;
 }
 
-inline bool isTracing(const at::ArrayRef<Variable>& vars) {
-  // Reference to avoid refcount bump
-  for (const Variable& var : vars) {
-    if (isTracing(var)) return true;
-  }
-  return false;
+inline bool isTracing(const at::Tensor& x) {
```

Comment:
can't it be template vararg?

Reply:
Yes, it could be. My template-fu is just not very good so I didn't implement it that way.